apps/forklift/lib/forklift/data_writer.ex

```diff
@@ -218,4 +223,18 @@ defmodule Forklift.DataWriter do
       dataset_id: dataset_id
     }
   end
+
+  defp create_ingestion_complete_data(dataset_id, ingestion_id, actual_message_count, extract_time) do
+    {expected_message_count, _} =
+      Redix.command!(:redix, ["GET", ingestion_id])
```

Comment:
I think this Redis key should be ingestion_id + extraction start time.

Just using the ingestion ID potentially leads to race conditions between ingestion triggers. If another ingestion starts before this one is complete for some reason, the expected message count would have a race condition.

This is more of a consideration if push-based ingestions were ever added (Which is likely a feature if UrbanOS ever handled IoT data). An ingestion could have 2 concurrent ingestions running.